### PR TITLE
fix #108

### LINKED ISSUE #108: @last_backlog should not be same as @backlog
## Pieces of codes in utils.rb
```ruby
DEBUG_MODE = true

    def debug_print msg
      print msg if DEBUG_MODE
    end

    RUBY = RbConfig.ruby
    REPL_RPOMPT = '(rdbg)'

    def create_pseudo_terminal(boot_options: "-r debug/run")
      inject_lib_to_load_path
      ENV['RUBY_DEBUG_USE_COLORIZE'] = "false"
      ENV['RUBY_DEBUG_TEST_MODE'] = 'true'

      timeout_sec = (ENV['RUBY_DEBUG_TIMEOUT_SEC'] || 10).to_i

      PTY.spawn("#{RUBY} #{boot_options} #{temp_file_path}") do |read, write, pid|
        @backlog = []
        @last_backlog = []
        ask_cmd = ['quit', 'delete', 'kill']
        begin
          Timeout.timeout(timeout_sec) do
            while (line = read.gets)
              debug_print line
              case line.chomp
              when /INTERNAL_INFO:\s(.*)/
                p ":last_backlog #{@last_backlog}"
                p ":backlog #{@backlog}"
                @internal_info = JSON.parse(Regexp.last_match(1))
                cmd = @queue.pop
                if cmd.is_a?(Proc)
                  cmd.call
                  cmd = @queue.pop

```

### Now
```shell
$ ruby test/debug/catch_test.rb
Loaded suite test/debug/catch_test
Started
[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb
=>    1| a = 1
      2| b = 2
      3|
      4| 1/0
=>#0	<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1
INTERNAL_INFO: {"location":"/var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1","line":1}
":last_backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1\\r\\n\"]"
":backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1\\r\\n\"]"
catch StandardError

(rdbg) catch StandardError
#0  BP - Catch  "StandardError"
INTERNAL_INFO: {"location":"/var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1","line":1}
":last_backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1\\r\\n\", \"catch StandardError\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) catch StandardError\\r\\n\", \"\\e[?2004l\\r#0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"
":backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1\\r\\n\", \"catch StandardError\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) catch StandardError\\r\\n\", \"\\e[?2004l\\r#0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"

(rdbg) continue
# No sourcefile available for /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb
=>#0	[C] Integer#/ at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:4
  #1	<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:4

Stop by #0  BP - Catch  "StandardError"
INTERNAL_INFO: {"location":"/var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:4","line":4}
":last_backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1\\r\\n\", \"catch StandardError\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) catch StandardError\\r\\n\", \"\\e[?2004l\\r#0  BP - Catch  \\\"StandardError\\\"\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) continue\\r\\n\", \"\\e[?2004l\\r# No sourcefile available for /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>#0\\t[C] Integer#/ at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:4\\r\\n\", \"  #1\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:4\\r\\n\", \"\\r\\n\", \"Stop by #0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"
":backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:1\\r\\n\", \"catch StandardError\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) catch StandardError\\r\\n\", \"\\e[?2004l\\r#0  BP - Catch  \\\"StandardError\\\"\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) continue\\r\\n\", \"\\e[?2004l\\r# No sourcefile available for /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb\\r\\n\", \"=>#0\\t[C] Integer#/ at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:4\\r\\n\", \"  #1\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66658-19rnmp.rb:4\\r\\n\", \"\\r\\n\", \"Stop by #0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"

(rdbg) q!

```

### Expected
```
$ ruby test/debug/catch_test.rb
Loaded suite test/debug/catch_test
Started
[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb
=>    1| a = 1
      2| b = 2
      3|
      4| 1/0
=>#0	<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:1
INTERNAL_INFO: {"location":"/var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:1","line":1}
":last_backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:1\\r\\n\"]"
":backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:1\\r\\n\"]"
catch StandardError

(rdbg) catch StandardError
#0  BP - Catch  "StandardError"
INTERNAL_INFO: {"location":"/var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:1","line":1}
":last_backlog [\"catch StandardError\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) catch StandardError\\r\\n\", \"\\e[?2004l\\r#0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"
":backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:1\\r\\n\", \"catch StandardError\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) catch StandardError\\r\\n\", \"\\e[?2004l\\r#0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"

(rdbg) continue
# No sourcefile available for /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb
=>#0	[C] Integer#/ at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:4
  #1	<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:4

Stop by #0  BP - Catch  "StandardError"
INTERNAL_INFO: {"location":"/var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:4","line":4}
":last_backlog [\"\\e[?2004h\\r\\n\", \"(rdbg) continue\\r\\n\", \"\\e[?2004l\\r# No sourcefile available for /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb\\r\\n\", \"=>#0\\t[C] Integer#/ at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:4\\r\\n\", \"  #1\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:4\\r\\n\", \"\\r\\n\", \"Stop by #0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"
":backlog [\"[1, 4] in /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb\\r\\n\", \"=>    1| a = 1\\r\\n\", \"      2| b = 2\\r\\n\", \"      3| \\r\\n\", \"      4| 1/0\\r\\n\", \"=>#0\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:1\\r\\n\", \"catch StandardError\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) catch StandardError\\r\\n\", \"\\e[?2004l\\r#0  BP - Catch  \\\"StandardError\\\"\\r\\n\", \"\\e[?2004h\\r\\n\", \"(rdbg) continue\\r\\n\", \"\\e[?2004l\\r# No sourcefile available for /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb\\r\\n\", \"=>#0\\t[C] Integer#/ at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:4\\r\\n\", \"  #1\\t<main> at /var/folders/kv/w1k6nh1x5fl7vx47b2pd005w0000gn/T/debugger20210619-66836-kkigg1.rb:4\\r\\n\", \"\\r\\n\", \"Stop by #0  BP - Catch  \\\"StandardError\\\"\\r\\n\"]"

(rdbg) q!
```

### PR DESCRIPTION
**Pull request by ono-max**

### CODE REVIEW

File: test/support/utils.rb [hunk 1]
```diff
@@ -64,6 +64,7 @@ def create_pseudo_terminal(boot_options: "-r debug/run")
                   cmd = @queue.pop
                 end
                 write.puts(cmd)
+                @last_backlog = []
```
Comment (st0012) [inline test/support/utils.rb:67]: perhaps Array#clear conveys the intention better?

**Review by st0012 — COMMENTED** [review]
Thanks for fixing this. I hit the same issue as well 😂

File: test/support/utils.rb (hunk 1 above)
Comment (ono-max) [reply test/support/utils.rb:67]: I think so, too. Thanks.

Comment (ono-max) [reply test/support/utils.rb]: https://github.com/ruby/debug/pull/109/files#diff-ee14bf65f14927f7778a246f90c5cec0c581be3e2658c2a8a3726c59db5be3f9R67

**Review by st0012 — APPROVED** [review]

**Pull request merged**